src/features/movies/MovieList/movieListSaga.js

```diff
@@ -1,36 +1,37 @@
-import { call, put, delay, takeLatest } from 'redux-saga/effects'
+import { call, put, delay, takeLatest, debounce } from 'redux-saga/effects';
 import {
     URLgenres,
     URLmovieSearch,
     URLpopularMovies,
-} from '../../../common/assets/generalData/fetchedData'
-import { setGenres } from './movieListSlice'
+} from '../../../common/assets/generalData/fetchedData';
+import { setGenres } from './movieListSlice';
 
-import { getData } from '../../getApiData'
+import { getData } from '../../getApiData';
 import {
     setMovieList,
     fetchMovieList,
     fetchMovieListError,
-} from './movieListSlice'
+} from './movieListSlice';
 
 function* fetchMovieListHandler({ payload: { query, page } }) {
     const searchedMovies = `${URLmovieSearch}&query=${query}&page=${page}`
     const popularMovies = `${URLpopularMovies}&page=${page}`
     try {
-        yield delay(1000)
-        const fetchedGenres = yield call(getData, URLgenres)
-        yield put(setGenres(fetchedGenres))
+        yield delay(500);
```

Comment:
are you sure 500 is a proper value? maybe we should increase the value up to even 2000 ?